app/javascript/controllers/proposal_form_controller.js

```diff
@@ -84,17 +84,16 @@ export default class extends Controller {
 
   latex () {
     let data = event.target.dataset
-
+    let _this = this
     if(data.value == 'all') {
       let proposalId = data.propid
-
-      $.post(`/submit_proposals?proposal=${proposalId}`, $('form#submit_proposal').serialize(), function(data) {
-        $.post("/proposals/" + proposalId + "/latex",
-          { latex: 'all' },
-          function(data, status) {
-            window.open(`/proposals/rendered_proposal.pdf`)
-        });
-      });
+      if(window.location.href.includes('edit')){
+        $.post(`/submit_proposals?proposal=${proposalId}`, 
+          $('form#submit_proposal').serialize(),
+          function() {
+            _this.renderPdf(proposalId)
+          });
+      } else this.renderPdf(proposalId)
```

Comment:
Codacy has a fix for the issue: [Expected { after 'else'.](https://app.codacy.com/gh/birs-math/proposals/pullRequest?prid=7656913)
```suggestion
      } else {this.renderPdf(proposalId)}
```